site/content/blog/2021-01-21-streamlining-developer-access-to-prometheus-and-grafana.md

```diff
@@ -0,0 +1,67 @@
+---
+title: "Streamlining Developer Access to Prometheus and Grafana"
+slug: streamlining-developer-access-to-prometheus-and-grafana
+date: 2021-01-21T00:00:00-04:00
+categories:
+    - "testing"
+    - "performance"
+author: Jesse Hallam
+github: lieut-data
+community: jesse.hallam
+summary: With access to the enterprise source code, the developer build tooling now automates the setup of Prometheus and Grafana for performance monitoring. Even the canonical Grafana dashboards are setup without any manual configuration required!
+---
+
+Our [Makefile](https://github.com/mattermost/mattermost-server/blob/master/Makefile) entrypoint for developing against the [Mattermost Server](https://github.com/mattermost/mattermost-server) already tries to simplify things for developers as much as possible.
+
+For example, when invoking `make run-server`, this build tooling takes care of all of the following (among other things!):
+* downloading a suitable version of `mmctl` for API-driven command line interaction
+* checking your installed Go version to avoid cryptic error messages
+* downloading and starting various Docker containers:
+    - `mysql` and `postgres`, both for a backend to the server and also to automate tests
+    - `inbucket` to simplify email testing without routing outside your machine
+    - `minio` to expose an S3-compatible interface to your local disk for for hosting uploaded files
+    - `openldap` to simplify testing alternate modes of authentication (requires access to the enterprise source)
+    - `elasticsearch` to expose an improved search backend for posts and channels (requires access to the enterprise source)
+* linking your server to an automatically detected `mattermost-webapp` directory
+* optionally starting the server in the foreground if `RUN_SERVER_IN_BACKGROUND` is `false`
+* piping the structured server logs through a `logrus` decorator for easier reading
+* invoking the `go` compiler to run the server code in your local repository
+
+### Adding Prometheus & Grafana
+
+As of https://github.com/mattermost/mattermost-server/pull/16649, this build tooling now supports automating the setup of Prometheus and Grafana Docker containers. Together, these tools form the backbone of our [Performance Monitoring (E20)](https://docs.mattermost.com/deployment/metrics.html) functionality, scraping metrics from Mattermost to help give enterprise customers insight into the performance of their clusters. Automating this setup makes it easier for developers to test in a production-like environment, and to help extend the monitoring available to customers.
+
+Since the corresponding Mattermost functionality is only available with access to the enterprise source code, these containers aren't enabled by default. To turn them on, export the following environment variable into your shell:
```

Comment:
```suggestion
Since the corresponding Mattermost functionality is only available with access to the Enterprise source code, these containers aren't enabled by default. To turn them on, export the following environment variable into your shell:
```